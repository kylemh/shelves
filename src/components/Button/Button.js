import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './Button.module.scss';

class Button extends Component {
  static propTypes = {
    children: PropTypes.oneOfType([PropTypes.node, PropTypes.string]).isRequired,
    className: PropTypes.string,
    disabled: PropTypes.bool,
    fullWidth: PropTypes.bool,
    onClick: PropTypes.func,
    tabIndex: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    theme: PropTypes.oneOf(['primary', 'secondary']),
    type: PropTypes.oneOf(['button', 'reset', 'submit']),
  };

  static defaultProps = {
    className: undefined,
    disabled: false,
    fullWidth: false,
    onClick: undefined,
    tabIndex: 0,
    theme: 'primary',
    type: 'button',
  };

  render() {
    const {
      children,
      className,
      disabled,
      fullWidth,
      onClick,
      tabIndex,
      theme,
      type,
    } = this.props;

    return (
      <button
        className={classNames(styles.Button, className, styles[theme], {
          [styles.fullWidth]: fullWidth,
        })}
        disabled={disabled}
        onClick={onClick}
        tabIndex={tabIndex}
        type={type}
      >
        {children}
      </button>
    );
  }
}

export default Button;
