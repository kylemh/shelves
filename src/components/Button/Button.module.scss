@import '../../styles/variables.scss';

.Button {
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  border: 2px solid transparent;
  border-radius: 10px;
  background-color: $white;
  font-family: $primaryFont;
  font-weight: bold;
  font-size: 1rem;
  vertical-align: middle;
  text-decoration: none;
  padding: 0.35rem;
  white-space: nowrap;
  transition:
    $animationDuration $animationTiming background-color,
    $animationDuration $animationTiming color;

  &:disabled {
    opacity: 0.6;

    &:hover {
      cursor: not-allowed;
    }
  }
}

.primary {
  background-color: rgba(0, 0, 0, 0);
  color: $primary;
  fill: $primary;
  border-color: $primary;
  outline-color: $primary;

  &:focus:enabled,
  &:hover:enabled {
    background-color: $primary;
    color: $white;
    fill: $white;
    outline-color: $white;
  }
}

.secondary {
  background-color: rgba(0, 0, 0, 0);
  color: $secondary;
  fill: $secondary;
  border-color: $secondary;

  &:focus:enabled,
  &:hover:enabled {
    background-color: $secondary;
    color: $white;
    fill: $white;
  }
}

.fullWidth {
  width: 100%;
}
