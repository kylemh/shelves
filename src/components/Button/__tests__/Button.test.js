import React from 'react';
import { mount, shallow } from 'enzyme';

import Button from '../Button';

describe('Button', () => {
  it('should render', () => {
    const wrapper = shallow(<Button>Test</Button>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call props.onClick when clicked', () => {
    const onClickMock = jest.fn();
    const wrapper = shallow(<Button onClick={onClickMock}>Test</Button>);

    expect(onClickMock).toHaveBeenCalledTimes(0);
    wrapper.simulate('click');
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  it('should not call props.onClick when clicked while disabled', () => {
    const onClickMock = jest.fn();
    const wrapper = mount(<Button onClick={onClickMock} disabled>Test</Button>);

    expect(onClickMock).toHaveBeenCalledTimes(0);
    wrapper.simulate('click');
    expect(onClickMock).toHaveBeenCalledTimes(0);
  });
});
