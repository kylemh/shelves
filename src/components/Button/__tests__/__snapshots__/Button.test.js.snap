// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render 1`] = `
<button
  className="Button primary"
  disabled={false}
  tabIndex={0}
  type="button"
>
  Test
</button>
`;
